<?xml version="1.0" encoding="utf-8"?>
<launch>
  <param name="/agentNum" value="1" />
  <rosparam command="load" file="$(find task_switch)/config/theta_1d.yaml" />
  <include file="$(find task_switch)/launch/joy.launch" />
  <node name="poseCollector" pkg="task_switch" type="posecollector.py" output="screen">
    <!-- for specify /bebop10*/virtualdrone/posestamped -->
    <!-- "preTopicName"bebop10*/"posestampedTopic" will be subscribed-->
    <param name="posestampedTopic" value="/pose" />
    <param name="preTopicName" value="/vrpn_client_node/" />
  </node>


  <node name="pcc_parameter" pkg="task_switch" type="task_switch_pcc_param_server" output="screen" />
  <node name="charge_parameter" pkg="task_switch" type="task_switch_charge_param_server" output="screen" />
  <node name="cbf_parameter" pkg="task_switch" type="task_switch_cbf_param_server" output="screen" />

  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" />

  <node pkg="task_switch" name="central" type="central_theta1d_node.py" output="screen">
    <param name="clock" value="20" />
  </node>


  <node pkg="task_switch" name="visualize_surf" type="pointcloud_theta1d.py" output="screen">
    <remap from="surf_value" to="/info" />
  </node>

  <include file="$(find task_switch)/launch/w8_pg_camera.launch">
    <arg name="view" value="false" />
  </include>

  <node pkg="rviz" type="rviz" name="rviz_pg" args="-d $(find task_switch)/config/pg_camera_view.rviz" />

  <rosparam command="load" file="$(find task_switch)/config/chargeset/bebop_all.yaml" />
  <include file="$(find task_switch)/launch/vrpn_client.launch" />

</launch>
